src/App.tsx

```diff
@@ -1,9 +1,26 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const handleChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const filterMovies = (movieList: Movie[]) => {
+    const searchValue = query.toLowerCase().trim();
+
+    return [...movieList].filter((movie) => {
+      return movie.title.toLocaleLowerCase().includes(searchValue)
+        || movie.description.toLocaleLowerCase().includes(searchValue);
+    });
+  };
```

Comment:
This can be moved outside of the component as a new reference will be created every re-render.